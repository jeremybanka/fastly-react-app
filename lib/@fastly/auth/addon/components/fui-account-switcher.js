import { OtpSetupError, PasswordRotateError } from '@fastly/auth/utils/errors'
import { set, setProperties } from '@ember/object'
import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // services
  session: service(),
  store: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  accounts: null,
  tagName: '',
  isRequestRunning: false,
  loading: false,
  error: null,

  // Methods
  // ---------------------------------------------------------------------------
  async fetchAccounts() {
    setProperties(this, {
      loading: true,
      didError: false,
    })

    const user = this.session.user

    const accounts = await this.store
      .query('account', { userId: user.id })
      .catch((err) => {
        console.error(err)
        set(this, 'error', 'fetchError')
      })
      .finally(() => {
        set(this, 'loading', false)
      })

    set(this, 'accounts', accounts)
  },
  /**
   * Switch the current session to the chosen account,
   * bailing early if the chosen account is the currentSession
   *
   * @param {Object} account
   * @param {string} account.id
   * @param {boolean} account.currentSession
   */
  handleChooseAccount({ id, currentSession }) {
    if (currentSession) return

    setProperties(this, {
      isRequestRunning: true,
      didError: false,
    })

    return this.session
      .switchAccount(id)
      .catch((err) => {
        if (err instanceof OtpSetupError) {
          set(this, 'error', 'otpSetupError')
        } else if (err instanceof PasswordRotateError) {
          set(this, 'error', 'passwordRotateError')
        } else {
          set(this, 'error', 'authError')
        }
      })
      .finally(() => {
        if (!this.isDestroyed && !this.isDestroying) {
          set(this, 'isRequestRunning', false)
        }
      })
  },

  handleKeyUp(account) {
    const ENTER_KEY_CODE = 13
    if (event.keyCode === ENTER_KEY_CODE) {
      this.handleChooseAccount(account)
    }
  },

  // Ember Hooks
  // ---------------------------------------------------------------------------
  init() {
    this.fetchAccounts()
    this._super(...arguments)
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <h1 class='fui-auth-screen__header'>
      Switch Accounts
    </h1>

    <p class='margin-bottom--m padding-x--m  text-align--left'>
      Switch to another account linked to {{session.user.login}}.
    </p>

    {{#if (eq error 'authError')}}
      <FuiAuthError />
    {{else if (eq error 'fetchError')}}
      <FuiAuthError
        @errorText='A problem occured while fetching accounts. Try again or'
      />
    {{else if (eq error 'otpSetupError')}}
      <FuiAuthError
        @errorText='2FA is required. To enable this now, visit your'
        @linkText='account settings'
        @link='/account/security'
      />
    {{else if (eq error 'passwordRotateError')}}
      <FuiAuthError
        @errorText='To switch accounts,'
        @linkText='reset your password'
        @link='/account/personal/password'
      />
    {{/if}}

  <ul class='fui-account-switcher__list list--outdented list--unbulleted padding--xs'>
    {{#each accounts as |account|}}
      <li
        tabindex='0'
        role='button'
        aria-current={{if account.currentSession 'true' false}}
        aria-disabled={{if account.currentSession 'true' false}}
        class='fui-account-switcher__list-item fui-card fui-card--accentuated fui-card--focus-ring flex-row flex-row--vcenter padding--m'
        onclick={{action handleChooseAccount account}}
        onkeyup={{action handleKeyUp account}}
      >
        <div class='flex-col--1 text-align--left'>
          <p class='head-bold'>{{account.name}}</p>
          <p class='color--grey'>{{account.id}}</p>
        </div>
        {{#if account.defaultAccount}}
          <div class='font--italic margin-x--l'>Default</div>
        {{/if}}
        <div class="fui-account-switcher__action">
          {{#if account.currentSession}}
            <span class='fui-tag fui-tag--emerald'>Signed In</span>
          {{else}}
            <span class='flex-col--pushed fui-button--link button'>
              Sign In
            </span>
          {{/if}}
        </div>
      </li>
    {{else}}
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
      <li
        class='fui-account-switcher__list-item fui-account-switcher__list-item--fixed-height fui-card fui-skeleton-overlay'
      />
    {{/each}}
  </ul>

  <p class='color--grey padding--s'>
    Don't see the account you're looking for?
    <ExternalLinkTo
      @text='Contact Support'
      @href='https://www.fastly.com/support'
    />.
  </p>
  `,
})
