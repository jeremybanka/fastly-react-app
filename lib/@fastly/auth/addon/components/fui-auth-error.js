import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  tagName: '',
  errorText: 'An error occured. Try again or ',
  linkText: 'Contact support',
  link: 'https://www.fastly.com/support',
  class: 'margin-bottom--m',
  dismissible: false,
  layout: hbs`
    <FuiMessage
      @type='error'
      @class='text-align--left {{class}}'
      @dismissible={{dismissible}}
    >
      <p role='alert' ...attributes >
        {{errorText}} <ExternalLinkTo
          @text={{linkText}}
          @href={{link}}
        />.
      </p>
    </FuiMessage>
  `,

  didInsertElement() {
    this._super(...arguments)
    const overlay = document.querySelector('.fui-auth-screen__overlay')
    if (overlay) {
      overlay.scrollTo({
        top: 0,
        behavior: 'smooth',
      })
    }
  },
})
