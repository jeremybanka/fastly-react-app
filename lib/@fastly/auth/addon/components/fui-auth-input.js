import Component from '@ember/component'
import { computed } from '@ember/object'
import { guidFor } from '@ember/object/internals'
import hbs from 'htmlbars-inline-precompile'

export default Component.extend({
  // Passed Properties
  // ---------------------------------------------------------------------------
  label: '',
  // Internal properties
  // ---------------------------------------------------------------------------
  tagName: '',
  inputId: computed(function () {
    return 'input-' + guidFor(this)
  }),

  layout: hbs`
    <div class='fui-auth-screen__input-group' aria-describedby={{tooltipId}}>
      <label class='fui-auth-screen__input-group-label flex-row margin-top--m margin-bottom--s' for={{inputId}}>
        {{label}}
        {{yield (hash contextual-link=(component 'yield'))}}
      </label>
      <Input @class='fui-input authentication__sign-in-input margin-top--s' @value={{value}} ...attributes id={{inputId}} />
    </div>
  `,
})
