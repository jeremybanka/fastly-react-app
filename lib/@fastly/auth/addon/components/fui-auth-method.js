import Component from '@ember/component'
import { computed } from '@ember/object'
import hbs from 'htmlbars-inline-precompile'
import { readOnly } from '@ember/object/computed'
import { inject as service } from '@ember/service'

export default Component.extend({
  tagName: '',
  session: service(),
  store: service(),

  accountName: computed('session.switchCustomerParams.customerId', function () {
    return this.store.peekRecord('account', this.session.switchCustomerParams.customerId).name
  }),

  ssoLoginUrl: readOnly('session.ssoLoginUrl'),
  layout: hbs`
    <h1 class="fui-auth-screen__header">Choose how to sign into <strong>{{accountName}}</strong></h1>
    <div class="padding-x--xxl padding-bottom--l">
      <ExternalLinkTo
        @href={{ssoLoginUrl}}
        class="button primary fui-login-form__button margin-bottom--l"
        aria-label="Sign in with single sign-on"
        target="_self"
      >
        Single sign-on
      </ExternalLinkTo>
      <LinkTo
        @route="switch-customer.login-required"
        class="button new fui-login-form__button margin-left--0"
        aria-label="Sign in with email and password"
      >
        Email and password
      </LinkTo>
    </div>
  `,
})
