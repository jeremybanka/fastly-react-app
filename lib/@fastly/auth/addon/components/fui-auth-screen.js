import Component from '@ember/component'
import hbs from 'htmlbars-inline-precompile'
import { inject as service } from '@ember/service'

export default Component.extend({
  // Services
  // ---------------------------------------------------------------------------
  router: service(),

  // Internal Properties
  // ---------------------------------------------------------------------------
  tagName: '',

  // Methods
  // ---------------------------------------------------------------------------
  cancel() {
    this.router.location.history.back()
  },

  // Template
  // ---------------------------------------------------------------------------

  layout: hbs`
    <div class='flex-row flex-row--hcenter text-align--center'>
      <div class='fui-card padding--l margin-top--xl'>
        <div class='fui-auth-screen__overlay'>
          <div class='fui-auth-screen__panel'>
            <FuiIcon @name='fastly-logo-red' @class='fastly-logo-red' />
            {{yield}}
            <button class='fui-auth-form__button cancel margin-top--m' {{action cancel}}>Cancel</button>
          </div>
        </div>
      </div>
    </div>
  `,
})
