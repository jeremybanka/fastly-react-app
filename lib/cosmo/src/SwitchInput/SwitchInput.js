/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Box, Flexbox, InputValue, Text } from "../";
import styled from "styled-components";
import css from "@styled-system/css";

function hexToRGBA(hex, alpha = 1) {
  const [r, g, b] = hex.match(/\w\w/g).map((x) => parseInt(x, 16));
  return `rgba(${r},${g},${b},${alpha});`;
}

function addOpacityToRGB(rgba, alpha = 1) {
  return `${rgba.slice(0, -1)}, ${alpha})`;
}

const StyledInput = styled("input")(() => {
  return css({
    // specificity hack to override legacy opacity
    "&&": {
      opacity: "0", // hides input and still allows :focus
    },
  });
});

const width = 64;
const height = 24;
const sliderSize = 32;
const sliderCheckedPosition = width - sliderSize;

const StyledInputContainer = styled("label")(() => {
  return css({
    position: "relative",
    display: "inline-block",
    width,
    height,
    verticalAlign: "middle",
  });
});

const StyledSwitch = styled("div")((props) => {
  return css({
    position: "absolute",
    top: "0",
    left: "0",
    right: "0",
    bottom: "0",
    backgroundColor: "grays.3",
    transition: "75ms",
    borderRadius: "10em",
    [`${StyledInput}:disabled + &`]: {
      backgroundColor: hexToRGBA(props.theme.colors.grays[3], 0.4),
      cursor: "not-allowed", // remove or apply to corresponding label?
    },
    [`${StyledInput}:checked:disabled + &`]: {
      backgroundColor: addOpacityToRGB(props.theme.colors.indigo, 0.4),
    },
    [`${StyledInput}:checked + &`]: {
      backgroundColor: "indigo",
    },
    "&:before": {
      boxSizing: "border-box",
      position: "absolute",
      content: '""',
      height: `${sliderSize}px`,
      width: `${sliderSize}px`,
      left: "0",
      bottom: `${height / 2 - sliderSize / 2}px`,
      backgroundColor: "grays.3",
      transition: "75ms",
      boxShadow: "lg",
      borderRadius: "10em",
      borderColor: "#fff", // force always white
      borderStyle: "solid",
      borderWidth: `${sliderSize / 2}px`,
    },
    [`${StyledInput}:disabled + &:before`]: {
      backgroundColor: hexToRGBA(props.theme.colors.grays[3], 0.4),
    },
    [`${StyledInput}:checked:disabled + &:before`]: {
      backgroundColor: addOpacityToRGB(props.theme.colors.indigo, 0.4),
    },
    [`${StyledInput}:focus + &:before, ${StyledInput}:active + &:before`]: {
      borderWidth: "8px", // theme.borderWidths / md
    },
    [`${StyledInput}:checked + &:before`]: {
      backgroundColor: "indigo",
      transform: `translateX(${sliderCheckedPosition}px)}`,
    },
  });
});

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type FieldInputProps = {
  name: string,
  id: string,
  // value?: string, // omit for `[name]=on`, include for `[name]=[value]`
  checked: boolean,
  // required?: boolean, // represent in UI?
  disabled?: boolean,
  onChange: (e: any) => void,
  onFocus?: (e: any) => void,
  onBlur?: (e: any) => void,
};

type SwitchInputProps = {
  input: FieldInputProps,
  label: TextType,
};

const SwitchInput = (props: SwitchInputProps) => {
  const { label, input } = props;
  return (
    <Box>
      <Flexbox alignItems="center">
        <Box>
          <StyledInputContainer htmlFor={input.id}>
            <StyledInput type="checkbox" {...input} />
            <StyledSwitch />
          </StyledInputContainer>
        </Box>
        <Box marginLeft="sm">
          <Box>
            <InputValue htmlFor={input.id}>{label}</InputValue>
          </Box>
        </Box>
      </Flexbox>
    </Box>
  );
};

export default SwitchInput;
