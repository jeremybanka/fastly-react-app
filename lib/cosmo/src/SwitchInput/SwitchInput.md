```jsx
<SwitchInput input={{}} label="Switcheroo" />
```
