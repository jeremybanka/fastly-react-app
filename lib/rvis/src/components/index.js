/* @flow */

import AxisX from "./AxisX";
import AxisY from "./AxisY";
import Background from "./Background";
import Brush from "./Brush";
import BrushActions from "./BrushActions";
import BrushMask from "./BrushMask";
import Chart from "./Chart";
import ChartInteractions from "./ChartInteractions";
import ChartMarker from "./ChartMarker";
import ChartMarkerText from "./ChartMarkerText";
import LegendTotal from "./LegendTotal";
import Loading from "./Loading";

export {
  AxisX,
  AxisY,
  Background,
  Brush,
  BrushActions,
  BrushMask,
  Chart,
  ChartInteractions,
  ChartMarker,
  ChartMarkerText,
  LegendTotal,
  Loading,
};
