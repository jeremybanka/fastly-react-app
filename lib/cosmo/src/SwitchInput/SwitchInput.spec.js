/* @flow */

import React from "react";
import { render, mount } from "enzyme";
import { InputValue } from "components";
import { ThemeProvider } from "styled-components";
import SwitchInput from "./SwitchInput";

const handleBlur = jest.fn();
const handleChange = jest.fn();
const basicInputObject = {
  name: "colors",
  id: "colors-blue",
  checked: true,
  onChange: handleChange,
  onBlur: handleBlur,
};

describe("SwitchInput test suite", () => {
  it("renders the stored snapshot", () => {
    const component = (
      <ThemeProvider
        theme={{
          colors: {
            grays: new Array(10).fill("#000"),
            indigo: "rgba(0, 0, 0)",
          },
        }}
      >
        <SwitchInput
          label="Blue"
          input={basicInputObject}
          helper="Could ancient greeks see blue?"
        />
      </ThemeProvider>
    );
    expect(render(component)).toMatchSnapshot();
  });

  it("renders with InputValue", () => {
    const component = (
      <ThemeProvider
        theme={{
          colors: {
            grays: new Array(10).fill("#000"),
            indigo: "rgba(0, 0, 0)",
          },
        }}
      >
        <SwitchInput label="Foo" input={basicInputObject} />
      </ThemeProvider>
    );
    expect(mount(component).find(InputValue)).toHaveLength(1);
  });
});
