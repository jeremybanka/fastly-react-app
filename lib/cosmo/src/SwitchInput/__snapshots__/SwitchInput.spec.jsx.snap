// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInput test suite renders the stored snapshot 1`] = `
<div
  class="Box-sc-1v2etov-0 kAhoak"
>
  <div
    class="Flexbox-sc-18yd7vq-0 dZrqmk"
  >
    <div
      class="Box-sc-1v2etov-0 kAhoak"
    >
      <label
        class="SwitchInput__StyledInputContainer-xapx83-1 lbDSnN"
        for="colors-blue"
      >
        <input
          checked=""
          class="SwitchInput__StyledInput-xapx83-0 cgzsJw"
          id="colors-blue"
          name="colors"
          type="checkbox"
        />
        <div
          class="SwitchInput__StyledSwitch-xapx83-2 hWaFXU"
        />
      </label>
    </div>
    <div
      class="Box-sc-1v2etov-0 fAwoJg"
    >
      <div
        class="Box-sc-1v2etov-0 kAhoak"
      >
        <label
          class="Text-sc-6c1xd4-0 jUjMgA"
          color="grays.6"
          font-size="xs"
          font-weight="semibold"
          for="colors-blue"
          letter-spacing="sm"
        >
          Blue
        </label>
      </div>
    </div>
  </div>
</div>
`;
